pkg/ipcache/kvstore.go

```diff
@@ -140,10 +140,13 @@ func (r *kvReferenceCounter) release(key string) (err error) {
 	defer r.Unlock()
 
 	refcnt, ok := r.keys[key] // 0 if not found
-	if ok {
-		refcnt--
+	// avoid underflow and report bug
+	if !ok || refcnt == 0 {
+		log.WithField("key", key).Error("BUG: attempt to release ipcache entry while refcnt == 0")
```

Comment:
Shouldn't this be added on https://github.com/cilium/cilium/blob/master/test/helpers/cons.go#L229?

Reply:
I think it makes sense to add `BUG` to that list, but in a separate PR. This is a bugfix, let's not mix in changes to code that might not be in an older release.